src/transaction/boot_sr.c

```diff
@@ -988,34 +994,50 @@ boot_make_temp_volume_fullname (char *temp_vol_fullname, VOLID temp_volid)
 }
 
 /*
- * boot_remove_unknown_temp_volumes () -
+ * boot_remove_unknown_temp_volumes () - to be called only upon server restart; unknown volumes means
+ *    volumes which are found existing (and should not be) vs the state that the server is at the
+ *    moment (ie: restart); performs the deletion in reverse order (aka LIFO) versus creation order
+ *    in order to keep the data consistent in case of abnormal server shutdown even when the boot
+ *    procedure is underway
  *
  * return: none
  */
 static void
 boot_remove_unknown_temp_volumes (THREAD_ENTRY * thread_p)
 {
-  VOLID temp_volid;
+  VOLID temp_volid = LOG_MAX_DBVOLID;
   char temp_vol_fullname[PATH_MAX];
 
-  if (boot_Db_parm->temp_last_volid == NULL_VOLID)
-    {
-      temp_volid = LOG_MAX_DBVOLID;
-    }
-  else
-    {
-      temp_volid = boot_Db_parm->temp_last_volid - 1;
-    }
+  assert (!BO_IS_SERVER_RESTARTED ());
+  /* logic below needs this invariant */
+  static_assert (LOG_MAX_DBVOLID < VOLID_MAX, "LOG_MAX_DBVOLID is expected to be less than VOLID_MAX");
 
-  for (; temp_volid > boot_Db_parm->last_volid; temp_volid--)
+  /* inspect volumes in descending order */
+  for (; temp_volid > LOG_DBFIRST_VOLID; temp_volid--)
     {
       boot_make_temp_volume_fullname (temp_vol_fullname, temp_volid);
       if (!fileio_is_volume_exist (temp_vol_fullname))
 	{
+	  /* the last found volume id */
+	  /* if no temporary volumes are found, and this is LOG_MAX_DBVOLID, incrementing here
+	   * will not overflow and will invalidate the loop below as well */
+	  ++temp_volid;
 	  break;
 	}
       er_set (ER_NOTIFICATION_SEVERITY, ARG_FILE_LINE, ER_BO_UNKNOWN_VOLUME, 1, temp_vol_fullname);
```

Comment:
ER_BO_UNKNOWN_VOLUME is also logged in the next loop. Remove a duplicate.
```suggestion
```

Reply:
Done